koku/masu/external/kafka_msg_handler.py

```diff
@@ -721,26 +702,46 @@ async def process_messages(msg, loop=EVENT_LOOP):
     return process_complete
 
 
-def get_consumer(event_loop):  # pragma: no cover
+def get_consumer():  # pragma: no cover
     """Create a Kafka consumer."""
-    return AIOKafkaConsumer(
-        HCCM_TOPIC,
-        loop=event_loop,
-        bootstrap_servers=Config.INSIGHTS_KAFKA_ADDRESS,
-        group_id="hccm-group",
-        enable_auto_commit=False,
+    consumer = Consumer(
+        {
+            "bootstrap.servers": Config.INSIGHTS_KAFKA_ADDRESS,
+            "group.id": "hccm-group",
+            "queued.max.messages.kbytes": 1024,
+            "enable.auto.commit": False,
+            "enable.auto.offset.store": False,
+            "max.poll.interval.ms": 1080000,  # 18 minutes
+        }
     )
+    consumer.subscribe([HCCM_TOPIC])
+    return consumer
 
 
-async def listen_for_messages_loop(event_loop):
+def listen_for_messages_loop():
     """Wrap listen_for_messages in while true."""
-    while True:
-        consumer = get_consumer(event_loop)
-        await listen_for_messages(consumer)
+    consumer = get_consumer()
+    LOG.info("Consumer is listening for messages...")
+    for _ in itertools.count():  # equivalent to while True, but mockable
+        msg = consumer.poll(timeout=1.0)
+        if msg is None:
+            continue
+
+        if msg.error():
+            KAFKA_CONNECTION_ERRORS_COUNTER.inc()
+            if msg.error().code() == KafkaError._PARTITION_EOF:
+                # End of partition event
+                LOG.warning(
+                    f"[listen_for_messages_loop] {msg.topic()} {msg.partition()} reached end at offset {msg.offset()}"
+                )
+                continue
+            elif msg.error():
+                raise KafkaException(msg.error())
```

Comment:
This will be game over for the listener?  I don't see this being caught in `koku_listener_thread`